src/QsCompiler/CompilationManager/DataStructures.cs

```diff
@@ -474,7 +475,7 @@ internal struct HeaderEntry<T>
             CodeFragment.TokenIndex tIndex,
             ImmutableArray<AttributeAnnotation> attributes,
             ImmutableArray<string> doc,
-            string? keepInvalid = null)
+            [AllowNull] string? keepInvalid = null)
```

Comment:
The `[AllowNull]` shouldn't be needed - `string?` should already allow null.

Reply:
I think I was getting a "fake" error. After merging in your changes with the nullable references and removing these, I am not getting the error I was getting before. I've removed this.